### PR TITLE
Build optimization

### PR DESCRIPTION
**Pull request by kiahjh**

### CODE REVIEW

File: apps/next-evans/lib/friend.ts [hunk 1]
```diff
@@ -14,6 +14,10 @@ export function getFriendUrl(friendSlug: string, gender: Gender): string {
   return `/${firstPart}/${friendSlug}`;
 }
 
+export function getDocumentUrl(friendSlug: string, documentSlug: string): string {
```
Comment (jaredh159) [inline apps/next-evans/lib/friend.ts:17]: 👍

File: apps/next-evans/lib/residences.ts [hunk 2]
```diff
@@ -43,6 +44,29 @@ function deriveMap(positions: Position[]): 'UK' | 'US' | 'Europe' {
   return arrays[0][0];
 }
 
+export function getPrimaryResidence(residences: Residence[]): Residence | null {
```
Comment (jaredh159) [inline apps/next-evans/lib/residences.ts:47]: help me remember that we need to dial this fully in, and test it, according to the swift code i showed you, at some point. doesn't need to hold up this PR though

File: apps/next-evans/next.config.js [hunk 3]
```diff
@@ -27,6 +27,9 @@ const nextConfig = {
       },
     ];
   },
+  eslint: {
+    ignoreDuringBuilds: true,
+  },
```
Comment (jaredh159) [inline apps/next-evans/next.config.js:32]: i think both of these edits were stuff we added while troubleshooting and experimenting, not stuff we wanted long term.

that easily may not have been clear to you... but if it was, these are the sorts of things you should catch (and maybe some of the process.stdout logging above) if when you do your PR self review 👍

File: apps/next-evans/pages/getting-started.tsx [hunk 4]
```diff
@@ -16,83 +14,25 @@ import { LANG } from '@/lib/env';
...
-export type GettingStartedCoverProps = Pick<
-  CoverProps,
-  'title' | 'author' | 'edition' | 'customCss' | 'customHtml'
-> & {
-  authorSlug: string;
-  hasAudio: boolean;
-  documentSlug: string;
-  authorGender: Gender;
-};
+export type GettingStartedCoverProps = Omit<
```
Comment (jaredh159) [inline apps/next-evans/pages/getting-started.tsx:32]: can you switch this to a `Pick`? the problem with `Omit` in this context, is if we add fields to our query, for some other page, this page will silently grow in size, without us wanting it to.

**Review by jaredh159 — CHANGES_REQUESTED** [review]
looks super good.  left a handful of comments for mostly minor things.  excellent work 👍

File: apps/next-evans/lib/db/friends.ts [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  }, {});
+}
+
+function addCustomCodeToFriends(
+  friends: Record<string, FriendProps>,
+  customCode: Record<string, CustomCode>,
+): Record<string, FriendProps> {
+  return Object.keys(friends).reduce<Record<string, FriendProps>>((acc, friendSlug) => {
+    const friend = friends[friendSlug];
+    invariant(friend);
```
[the remark this answers is not in the document]
Comment (kiahjh) [reply apps/next-evans/lib/db/friends.ts:177]: I don't think that works, `invariant` returns undefined

Screenshot 2023-06-13 at 9 15 01 AM

File: apps/next-evans/next.config.js (hunk 3 above)
Comment (kiahjh) [reply apps/next-evans/next.config.js:32]: Gotcha, I missed that

**Review by jaredh159 — APPROVED** [review]

**Pull request merged**